lib/ansible/plugins/inventory/tower.py

```diff
@@ -159,7 +164,18 @@ def parse(self, inventory, loader, path, cache=True):
                                   force_basic_auth=True,
                                   validate_certs=self.get_option('validate_certs'))
 
-        inventory_id = self.get_option('inventory_id').replace('/', '')
+        # validate type of inventory_id because we allow two types as special case
+        inventory_id = self.get_option('inventory_id')
+        if isinstance(inventory_id, int):
+            inventory_id = to_native(self.get_option('inventory_id'))
```

Comment:
Since this is controller code, this should probably be using ```to_text()``` here.  The ```self.get_option```  also seems redundant since you just pulled that value into inventory_id above?  So maybe:
```suggestion
            inventory_id = to_text(inventory_id)
```

Reply:
(Using nonstring is not necessary as simplerepr is the default but I like to use it to flag this as a case where we're using the function to get the textual representation of an object rather than a case where we're converting from bytes to text.)

Reply:
> Since this is controller code, this should probably be using to_text() here.

What is the full context of "here"?

Should all existing occurrences of `to_native` (4) in this file be changed to `to_text`? This diff constitutes 2 of those 4 occurrences. The type of use does not differ from the new code vs. prior code. Because of that, my preferred interpretation of your comment is to replace them all.

Will push commit with that change.

Reply:
We talked this over on slack and the new commits look good now.